FirebasePerformance/Sources/Instrumentation/FPRNetworkTrace.m

```diff
@@ -137,14 +140,20 @@ - (NSString *)description {
   return [NSString stringWithFormat:@"Request: %@", _URLRequest];
 }
 
-- (void)updateTraceWithCurrentSession {
+- (void)sessionChanged:(NSNotification *)notification {
   if (self.traceStarted && !self.traceCompleted) {
-    dispatch_async(self.sessionIdSerialQueue, ^{
-      FPRSessionManager *sessionManager = [FPRSessionManager sharedInstance];
-      FPRSessionDetails *sessionDetails = [sessionManager.sessionDetails copy];
-      if (sessionDetails) {
-        [self.activeSessions addObject:sessionDetails];
-      }
+    NSDictionary<NSString *, FPRSessionDetails *> *userInfo = notification.userInfo;
+    FPRSessionDetails *sessionDetails = [userInfo valueForKey:kFPRSessionIdNotificationKey];
+    if (sessionDetails) {
+      [self updateTraceWithCurrentSession:sessionDetails];
+    }
+  }
+}
+
+- (void)updateTraceWithCurrentSession:(FPRSessionDetails *)sessionDetails {
+  if (sessionDetails != nil) {
+    dispatch_sync(self.sessionIdSerialQueue, ^{
```

Comment:
Does it have to be `dispatch_sync`? Using `dispatch_sync` usually increases probability to introduce a deadlock unintentionally so should be avoided when possible.

Reply:
I thought quite a bit about this and preferred to change this to sync (had it as async earlier). Agree with the possibility of the deadlock situation. Since this queue is internally handled and it not used for any purpose other than queueing the addition of sessionIds, I felt it is safe to go with a synchronous operation for this.